app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -1,133 +1,214 @@
+require 'uri'
 require 'websocket'
 require 'rubygems/package'
 
+require 'active_support/time'
+require 'websocket-client-simple'
+
+module WebSocket
+  module Client
+    module Simple
+      class Client
+        alias_method :send_msg, :send
+      end
+    end
+  end
+end
+
 module Authentication
   module AuthnK8s
-    class KubectlExec
-      STDIN_CHANNEL = 0
-      STDOUT_CHANNEL = 1
-      STDERR_CHANNEL = 2
-      ERROR_CHANNEL = 3
-      RESIZE_CHANNEL = 4
+    CommandTimedOut = ::Util::ErrorClass.new(
+      "Command timed out in container '{0}' of pod '{1}'"
+    )
+
+    class Messages
+      attr_reader :messages
+      
+      def initialize
+        @messages = Hash.new { |hash,key| hash[key] = [] }
+      end
+
+      def save_message(msg)
+        strm ||= stream_name(msg)
```

Comment:
Coming back to this code I wrote, I found the "channel" and "stream" names confusing and not immediately clear.  I looked at the docs around websockets and event machine, and didn't find any mention of "stream" in the sense i've used it here.  I think we should change to "channel_name" and "channel_number" terminology, unless there is something even more correct.  but those at names at least make clear the essential distinction in the context of this code.

Reply:
Yeah this was super confusing when I was trying to find that bug. Sounds good to me.